pkgdb/src/versions.cc

```diff
@@ -232,18 +208,14 @@ cleanRange( const std::string & range )
 std::list<std::string>
 semverSat( const std::string & range, const std::list<std::string> & versions )
 {
-  std::list<std::string> args
-    = { "--include-prerelease", "--loose", "--range", cleanRange( range ) };
-  for ( const auto & version : versions ) { args.push_back( version ); }
-  auto [ec, lines] = runSemver( args );
-  /* TODO: determine parse error vs. empty list result. */
-  if ( ! nix::statusOk( ec ) ) { return {}; }
```

Comment:
It's possible this still needs to have an analogous conditional i the new variant?

Reply:
I don't think so, at least not to preserve the existing functionality?
Previously, if there was a functional problem determining if it's satisfied (non-OK exit code), we returned an empty list.  In the new we check for validity, and if it fails, we'll simply not include it, potentially leading to an empty list to be returned, same as before.  Is there a different condition that you had in mind?  

We could enhance the functionality by catching/throwing to inform the upper layers of malformed versions / ranges.  But in looking upstream, there seems to be code that checks for valid ranges, makes some assumptions, tries to do the right thing.  And so I'm not too inclined to start mucking with a bunch of that.